common/src/Microsoft.Azure.IIoT.Serializers.NewtonSoft/src/Default/NewtonSoftJsonSerializer.cs

```diff
@@ -67,14 +80,31 @@ public class NewtonSoftJsonSerializer : IJsonSerializerSettingsProvider,
         }
 
         /// <inheritdoc/>
-        public object Deserialize(ReadOnlyMemory<byte> buffer, Type type) {
+        public object Deserialize(ReadOnlyMemory<byte> buffer, Type type, TextReader schemaReader = null) {
             try {
+
                 // TODO move to .net 3 to use readonly span as stream source
-                var jsonSerializer = JsonSerializer.CreateDefault(Settings);
-                using (var stream = new MemoryStream(buffer.ToArray()))
-                using (var reader = new StreamReader(stream, ContentEncoding)) {
-                    return jsonSerializer.Deserialize(reader, type);
+                var bufferArray = buffer.ToArray();
+
+                // Validate json if schema is provided.
+                if (schemaReader != null) {
+                    var validationResults = _jsonSchemaValidator.Validate(bufferArray, schemaReader);
+
+                    // If any item in validation results collection is marked as invalid, consider it as json schema validation failure.
+                    if (validationResults.Any(r => r.IsValid == false)) {
+                        var validationResultsMessage =
+                            string.Join("; ",
+                                validationResults.Select(r =>
```

Comment:
I think that we need to print messages only from the results that are invalid, meaning we need to first filter on `(r => !r.IsValid)` before joining.

Reply:
Can we please have a test as well that will make sure that only messages from invalid validation results are printed ?

Reply:
Thanks, that's a valid point, and this is being checked in test called `EnsureMultipleValidationErrorsAreFlattened` i.e. we throw multiple validation errors in this test and validate that all items in `validationResults` are invalid. So, no valid messages are present in collection when schema validation fails (for single or multiple errors) but I think I agree, having a check here would be more reassuring if the underlying implementation change in future.

Reply:
Plus checked in `PnPlcSchemaValidationFailureTest` test from deserializer perspective.

Reply:
@suneetnangia, I wanted to come up with a small example that would have a valid object and then an invalid one to see if the valid one is present in the result or not, so I modified `ConfigurationFileWithLargeErrorListIsHandledSuccessfully()` test and provided it with the following content:

```json
[
  {
    "EndpointUrl": "opc.tcp://20.185.195.172:53530/OPCUA/SimulationServer",
    "UseSecurity": false,
    "OpcNodes": [
      {
        "Id": "i=1001",
        "OpcSamplingInterval": 2000,
        "OpcPublishingInterval": 5000,
      },
      {
        "OpcSamplingInterval": 2000,
        "OpcPublishingInterval": 5000,
      }
    ]
  }
]
```

As you can see the second object in `OpcNodes` list does not have an `Id` property. And to my surprise validation result was that this was a valid content for published nodes file. Returned array contained only one element that had its `IsValid` property equal to `true`. Can you please have a look at this sample and see why is it not reporting an invalid content here and complaining that second element should have an `Id` field ?

Reply:
Investigating... on the face of it we probably need to fix this in Json schema (cc @WilliamBerryiii ), added additional test to cover this, thanks for highlighting it.

Reply:
I'm working to modify my schema generation tool to improve this issue and will additionally add a battery of tests to cover various conditions once the schema is updated.

Reply:
Schema changes will be pushed in next commit.

Reply:
Thanks for fixing this.